repo_blob.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2015 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package git
+
+// CatFileBlob returns the blob corresponding to the given revision of the repository.
+func (repo *Repository) CatFileBlob(rev string) (*Blob, error) {
+	typ, err := repo.CatFileType(rev)
+	if err != nil {
+		return nil, err
+	}
+
+	if typ != ObjectBlob {
+		return nil, ErrNotBlob
+	}
+
+	return &Blob{
+		TreeEntry: &TreeEntry{
+			mode: EntryBlob,
+			typ:  ObjectBlob,
+			id:   MustIDFromString(rev),
```

Comment:
There is no guarantee the `rev` being passed here is always a full-length SHA, I think we need to same thing as in `CatFileCommit`:

https://github.com/gogs/git-module/blob/a2567c624d4263a01df4a5405ebe065a9f96e0c3/repo_commit.go#L95-L98